akka-remote/src/main/scala/akka/remote/RemoteWatcher.scala

```diff
@@ -287,15 +287,9 @@ private[akka] class RemoteWatcher(
    * does not exist.
    */
   def reWatch(address: Address): Unit =
```

Comment:
What shall we do the first time, when uid is not know?
The uid is of importance to trigger quarantine.

Triggering DeathWatchNotification upon uid change when the previous uid was know might work, but what makes that work now and not previously? Just curious to understand if I missed something.